pkg/storage/chunk/fetcher/fetcher.go

```diff
@@ -138,6 +140,10 @@ func (c *Fetcher) FetchChunks(ctx context.Context, chunks []chunk.Chunk) ([]chun
 	l2OnlyChunks := make([]chunk.Chunk, 0, len(chunks))
 
 	for _, m := range chunks {
+		if c.skipQueryWritebackCacheOlderThan > 0 && m.From.Time().Before(time.Now().UTC().Add(-c.skipQueryWritebackCacheOlderThan)) {
```

Comment:
this does not look like the right place for this check. It is preventing the fetcher from fetching chunks older than `skipQueryWritebackCacheOlderThan`.

`WriteBackCache()` method might be a better place to add this

Reply:
Thank you for catching that and I do have the check in the WriteBackCache. 

What I do want to add is the ability to short circuit a cache lookup if it before this duration value. There might already be a setting for that and need to take a deeper look

Reply:
@ashwanthgoli This is going to fit our needs for the time being and would like to get this merged. 

If you feel there should be some work to be done to short-circuit the read path for these requests to the cache, we can look to add that.

Reply:
> What I do want to add is the ability to short circuit a cache lookup if it before this duration value. 

Sorry, I think you go it right the first time. [processCacheResponse](https://github.com/grafana/loki/blob/51471d1b363e655c2ae976c5172a35a3520d9bcc/pkg/storage/chunk/fetcher/fetcher.go#L188) should be able to figure out the missing keys and request them from store using the `chunks` request

Reply:
rest looks good! Happy to approve once you revert the latest commit